src/WidgetBase.ts

```diff
@@ -71,6 +69,23 @@ export function diffProperty(propertyName: string, diffType = DiffType.CUSTOM, d
 }
 
 /**
+ * Decorator that can be used to enable property diffing of all properties
+ */
+export function diffProperties(constructor: Function): void;
```

Comment:
Do we need this? Is it unreasonable to have the consumers *always* call `@diffProperties()`?